### DISCUSSION TITLE
0.4.0 Local Mode Available

### DISCUSSION THREAD

**Discussion by linonetwo** [tiddly-gittly/TidGi-Desktop#49] [Release]
Been a long time guys! I'm currently working at ByteDance Inc, and because we don't have 996 in our group, besides playing [Cataclysm-DDA](https://github.com/CleverRaven/Cataclysm-DDA), I have been always updating this project after work! Here are the update logs:

久等了各位！我现在在字节跳动打工，由于我们组没有 996，所以工作之余除了玩 [Cataclysm-DDA](https://github.com/CleverRaven/Cataclysm-DDA) 以外，我还有一直在更新这个项目！以下是本次的更新内容：

## Plugin 插件

No plugins updated this time. But the next update will focus on plugins!

这次没有插件更新， 但下次更新会把重心放在插件上！

## Feature 功能

- Rewrite using typescript, so there might be fewer bugs, hopefully.
- You can choose to create "Local Workspace", TiddlyGit will still create a local git backup system for you, and not Github connection is required!
- Expose some API for plugins to create a context menu, this will enable some fancy features on tw plugins, e.g., add tag via right click context menu.
- Git and Wiki now all run in separate worker threads, so App will run faster.
- UI and Icon update, match the latest macOS design.

- 使用typescript重写，所以可能会有更少的bug，希望如此。
- 你可以选择创建 「本地工作区」，太记仍将为你创建一个本地 git 备份系统，好处是不再强依赖 Github 了！
- 为插件暴露一些 API，可以用来创建上下文菜单，这将使得 tw 插件可以实现一些花哨的功能，比如通过右键菜单添加标签等等。
- Git 和 Wiki 现在都在独立的工作线程中运行，所以 App 会运行得更快。
- UI 和图标更新，符合最新的 macOS 设计。

## Translations

Now basically everything has an English translation.

现在基本所有文本都有中英翻译了。还有，我们选择将 TiddlyWiki 翻译成「太微」，TiddlyGit 翻译成「太记」，希望这个译名能帮助 TiddlyWiki 更好地触达到更多新用户。

<hr /><em>This discussion was created from the release <a href='https://github.com/tiddly-gittly/TiddlyGit-Desktop/releases/tag/v0.4.0'>0.4.0 Local Mode Available</a>.</em>

**Comment by ubuntupunk** [comment]
How does one set default language to English?

**Reply by linonetwo** [reply]
Hi, in TG, you can switch language in the language/menu

<img width="103" alt="截屏2021-10-23 22 12 46" src="https://user-images.githubusercontent.com/3746270/138560084-704dce9f-c8c2-437f-bed3-7723d663c9c1.png">

In TW, you can install an English language plugin, https://tiddlywiki.com/#Languages

**Comment by ubuntupunk** [comment]
Thanks I now found it.

**Reply by linonetwo** [reply]
I will make it easier to change language in the future, by provide a landing page that can choose language
